src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -732,14 +739,21 @@ def get_events(
             if until_interval
             else ""
         )
+        first_clause = f"limit {first}" if first else ""
+        last_clause = f"limit {last}" if last else ""
         query = dedent(
             f"""\
-            select timestamp, value::varchar value
-            from {self.account_event_table}
-            where resource_attributes:"snow.database.name" = '{app_name}'
-            {since_clause}
-            {until_clause}
-            order by timestamp asc;"""
+            select * from (
+                select timestamp, value::varchar value
+                from {self.account_event_table}
+                where resource_attributes:"snow.database.name" = '{app_name}'
+                {since_clause}
+                {until_clause}
+                order by timestamp desc
+                {last_clause}
+            ) order by timestamp asc
+            {first_clause}
```

Comment:
have you checked what happens to the plans when we do this? Do we need to worry about that cases where it's not strictly required?

Reply:
I compared
```sql
explain
select *
from build_finetune_db.public.events
where resource_attributes:"snow.database.name" = 'PYTHON_CALLBACKS_FCAMPBELL'
order by timestamp asc
```
and
```sql
explain
select * from (
    select *
    from build_finetune_db.public.events
    where resource_attributes:"snow.database.name" = 'PYTHON_CALLBACKS_FCAMPBELL'
    order by timestamp desc
    -- limit 10
)
order by timestamp asc
```
and the plan is exactly the same:
```
GlobalStats:
    partitionsTotal=812
    partitionsAssigned=2
    bytesAssigned=19659776
Operations:
1:0     ->Result  EVENTS.TIMESTAMP, EVENTS.START_TIMESTAMP, EVENTS.OBSERVED_TIMESTAMP, EVENTS.TRACE, EVENTS.RESOURCE, EVENTS.RESOURCE_ATTRIBUTES, EVENTS.SCOPE, EVENTS.SCOPE_ATTRIBUTES, EVENTS.RECORD_TYPE, EVENTS.RECORD, EVENTS.RECORD_ATTRIBUTES, EVENTS.VALUE, EVENTS.EXEMPLARS  
1:1          ->Sort  EVENTS.TIMESTAMP ASC NULLS LAST  
1:2               ->Filter  (TO_CHAR(GET_PATH(EVENTS.RESOURCE_ATTRIBUTES, '[''snow.database.name'']'))) = 'PYTHON_CALLBACKS_FCAMPBELL'  
1:3                    ->TableScan  BUILD_FINETUNE_DB.PUBLIC.EVENTS  TIMESTAMP, START_TIMESTAMP, OBSERVED_TIMESTAMP, TRACE, RESOURCE, RESOURCE_ATTRIBUTES, SCOPE, SCOPE_ATTRIBUTES, RECORD_TYPE, RECORD, RECORD_ATTRIBUTES, VALUE, EXEMPLARS  {partitionsTotal=812, partitionsAssigned=2, bytesAssigned=19659776}
```

So when there's no `--last`, the `order by timestamp desc` is completely optimized out

Reply:
Awesome, thanks for checking